sound/soc/sof/compress.c

```diff
@@ -287,6 +287,9 @@ static int sof_compr_trigger(struct snd_soc_component *component,
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_RELEASE;
 		break;
+	case SNDRV_PCM_TRIGGER_DRAIN:
+		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_DRAIN;
+		break;
```

Comment:
you probably need a different behavior for capture and playback.

From the alsa-lib PCM documentation https://www.alsa-project.org/alsa-doc/alsa-lib/group___p_c_m.html#ga49afc5b8527f30c33fafa476533c9f86

"
For playback wait for all pending frames to be played and then stop the PCM. For capture stop PCM permitting to retrieve residual frames.
"

Reply:
Yeah, most likely so